packages/agreement-process/src/services/agreementActivationProcessor.ts

```diff
@@ -67,7 +67,7 @@ export async function activateAgreementLogic(
   storeFile: FileManager["storeBytes"],
   correlationId: string,
   logger: Logger
-): Promise<Array<CreateEvent<AgreementEvent>>> {
+): Promise<[Agreement, Array<CreateEvent<AgreementEvent>>]> {
```

Comment:
may I ask you why an array and not an object?

Reply:
In this case, I preferred to use a tuple instead of an entire object. I found that it was more readable and more concise.